src/uucore/src/lib/mods/backup_control.rs

```diff
@@ -37,22 +33,164 @@ pub fn determine_backup_suffix(supplied_suffix: Option<&str>) -> String {
-                "none" | "off" => BackupMode::NoBackup,
-                _ => panic!(), // cannot happen as it is managed by clap
-            },
+/// Determine the "mode" for the backup operation to perform, if any.
+///
+/// Parses the backup options according to the [GNU manual][1], and converts
+/// them to an instance of `BackupMode` for further processing.
+///
+/// For an explanation of what the arguments mean, refer to the examples below.
+///
+/// [1]: https://www.gnu.org/software/coreutils/manual/html_node/Backup-options.html
+///
+///
+/// # Errors
+///
+/// If an argument supplied directly to the long `backup` option, or read in
+/// through the `VERSION CONTROL` env var is ambiguous (i.e. may resolve to
+/// multiple backup modes) or invalid, an error is returned. The error contains
+/// the formatted error string which may then be passed to the
+/// [`show_usage_error`] macro.
+///
+///
+/// # Examples
+///
+/// Here's how one would integrate the backup mode determination into an
+/// application.
+///
+/// ```
+/// let OPT_BACKUP: &str = "backup";
+/// let OPT_BACKUP_NO_ARG: &str = "b";
+/// let matches = App::new("myprog")
+///     .arg(Arg::with_name(OPT_BACKUP_NO_ARG)
+///         .short(OPT_BACKUP_NO_ARG)
+///     .arg(Arg::with_name(OPT_BACKUP)
+///         .long(OPT_BACKUP)
+///         .takes_value(true)
+///         .require_equals(true)
+///         .min_values(0))
+///     .get_matches_from(vec![
+///         "myprog", "-b", "--backup=t"
+///     ]);
+///
+/// let backup_mode = backup_control::determine_backup_mode(
+///     matches.is_present(OPT_BACKUP_NO_ARG), matches.is_present(OPT_BACKUP),
+///     matches.value_of(OPT_BACKUP)
+/// );
+/// let backup_mode = match backup_mode {
+///     Err(err) => {
+///         show_usage_error!("{}", err);
+///         return 1;
+///     },
+///     Ok(mode) => mode,
+/// };
+/// ```
+///
+/// This example shows an ambiguous imput, as 'n' may resolve to 4 different
+/// backup modes.
+///
+///
+/// ```
+/// let OPT_BACKUP: &str = "backup";
+/// let OPT_BACKUP_NO_ARG: &str = "b";
+/// let matches = App::new("myprog")
+///     .arg(Arg::with_name(OPT_BACKUP_NO_ARG)
+///         .short(OPT_BACKUP_NO_ARG)
+///     .arg(Arg::with_name(OPT_BACKUP)
+///         .long(OPT_BACKUP)
+///         .takes_value(true)
+///         .require_equals(true)
+///         .min_values(0))
+///     .get_matches_from(vec![
+///         "myprog", "-b", "--backup=n"
+///     ]);
+///
+/// let backup_mode = backup_control::determine_backup_mode(
+///     matches.is_present(OPT_BACKUP_NO_ARG), matches.is_present(OPT_BACKUP),
+///     matches.value_of(OPT_BACKUP)
+/// );
+/// let backup_mode = match backup_mode {
+///     Err(err) => {
+///         show_usage_error!("{}", err);
+///         return 1;
+///     },
+///     Ok(mode) => mode,
+/// };
+/// ```
+pub fn determine_backup_mode(
+    sopt_present: bool,
+    lopt_present: bool,
+    lopt_value: Option<&str>,
```

Comment:
nit: could you use a longer name for `sopt` and `lopt`, like `long_opt` and `short_opt`? That's a bit more obvious when reading the code.

Reply:
Done!